scripts/rddepman.ts

```diff
@@ -0,0 +1,145 @@
+import { WSLDistro, HostResolverHost, HostResolverPeer } from 'scripts/dependencies/wsl';
+import {
+  DependencyVersions, readDependencyVersions, writeDependencyVersions, Dependency, AlpineLimaISOVersion, getOctokit,
+} from 'scripts/lib/dependencies';
+
+const MAIN_BRANCH = 'main';
+const GITHUB_OWNER = 'rancher-sandbox';
+const GITHUB_REPO = 'rancher-desktop;';
+
+type VersionComparison = {
+  name: string;
+  currentVersion: string | AlpineLimaISOVersion;
+  latestVersion: string | AlpineLimaISOVersion;
+};
+
+const dependencies: Dependency[] = [
+  new tools.KuberlrAndKubectl(),
+  new tools.Helm(),
+  new tools.DockerCLI(),
+  new tools.DockerBuildx(),
+  new tools.DockerCompose(),
+  new tools.DockerProvidedCredHelpers(),
+  new tools.Trivy(),
+  new tools.GuestAgent(),
+  new tools.Steve(),
+  new tools.RancherDashboard(),
+  new tools.ECRCredHelper(),
+  new LimaAndQemu(),
+  new AlpineLimaISO(),
+  new WSLDistro(),
+  new HostResolverHost(),
+  new HostResolverPeer(),
+  new MobyOpenAPISpec(),
+];
+
+function git(...args: string[]): number | null {
+  const result = spawnSync('git', args);
+
+  if (result.error) {
+    throw result.error;
+  }
+
+  return result.status;
+}
+
+async function createDependencyBumpPR(name: string, currentVersion: string | AlpineLimaISOVersion, latestVersion: string | AlpineLimaISOVersion): Promise<void> {
+  const title = `rddepman: bump ${ name } from ${ currentVersion } to ${ latestVersion }`;
+  const branchName = `rddepman-bump-${ name }-from-${ currentVersion }-to-${ latestVersion }`;
+
+  await getOctokit().rest.pulls.create({
+    owner: GITHUB_OWNER,
+    repo:  GITHUB_REPO,
+    title,
+    base:  MAIN_BRANCH,
+    head:  branchName,
+  });
+}
+
+async function checkDependencies(): Promise<void> {
+  // load current versions of dependencies
+  const depVersionsPath = path.join('src', 'assets', 'dependencies.yaml');
+  const currentVersions = await readDependencyVersions(depVersionsPath);
+
+  // Get a list of dependencies for which:
+  // a) current version !== latest version
+  // b) there is not already a PR (open or closed) to perform this exact bump
+  const versionUpdates: VersionComparison[] = [];
+  const promises = dependencies.map(async(dependency) => {
+    const latestVersion = await dependency.getLatestVersion();
+    const currentVersion = currentVersions[dependency.name as keyof DependencyVersions];
+    const name = dependency.name;
+
+    if (JSON.stringify(currentVersion) === JSON.stringify(latestVersion)) {
+      console.log(`Dependency "${ name }" is at latest version "${ JSON.stringify(currentVersion) }".`);
+
+      return;
+    }
+
+    // try to find PR for this combo of name, current version and latest version
+    const branchName = `rddepman-bump-${ name }-from-${ currentVersion }-to-${ latestVersion }`;
+
+    try {
+      const response = await getOctokit().rest.pulls.list({
+        owner: GITHUB_OWNER, repo: GITHUB_REPO, base: branchName,
```

Comment:
Isn't this looking for PRs in the wrong direction? It's listing PRs that have a _base_ of `rddepman-bump-…` (i.e. things that will be merged _into_ the bump branch), rather than a base of `main` (i.e. the normal flow).

Reply:
Right you are, fixed.